devicetypes/smartthings/orvibo-Moisture-Sensor.src/orvibo-Moisture-Sensor.groovy

```diff
@@ -0,0 +1,192 @@
+		} else if(description?.startsWith('enroll request')){
+			List cmds = zigbee.enrollResponse()
+			log.debug "enroll response: ${cmds}"
+			result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+		}else {
+			Map descMap = zigbee.parseDescriptionAsMap(description)
+			if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0002) {
+				def zs = new ZoneStatus(zigbee.convertToInt(descMap.value, 16))
+				map = zs.isAlarm1Set() ? getMoistureResult('wet') : getMoistureResult('dry')
+				result = createEvent(map)
+			} else if (descMap?.clusterInt == 0x0001 && descMap?.commandInt != 0x07 && descMap?.value) {
+				if(descMap?.attrInt == 0x0021){
+					map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+					result = createEvent(map)
+				}else if(descMap?.attrInt == 0x0020){
+					map = getBatteryResult(Integer.parseInt(descMap.value, 16))
+					result = createEvent(map)
+				}
+			}
+		}
+	}
+
+	log.debug "Parse returned $result"
+
+	result
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	log.debug "Refreshing Values"
+	def refreshCmds = []
+	//the device support two way to get the battery value
+	refreshCmds += zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021)
+	refreshCmds += zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020)
+	refreshCmds += zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS) +
+		zigbee.enrollResponse()
+
+	refreshCmds
+}
+
+def configure() {
+	sendEvent(name: "checkInterval", value: 20 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	log.debug "Configuring Reporting"
+	def configCmds = []
+	configCmds += zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021, DataType.UINT8, 30, 21600, 0x10)
+	//	configCmds += zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020, DataType.UINT8, 30, 21600, 0x10)
+	refresh() + configCmds // send refresh cmds as part of config
+}
+
+def getMoistureResult(value) {
+	log.debug "water"
+	def descriptionText
+	if (value == "wet")
+		descriptionText = "${device.displayName} is wet"
+	else
+		descriptionText = "${device.displayName} is dry"
+	[
+		name           : 'water',
+		value          : value,
+		descriptionText: descriptionText,
+		translatable   : true
+	]
+}
+private Map getBatteryPercentageResult(rawValue) {
+	log.debug "Battery Percentage"
+	def result = [:]
+
+	if (0 <= rawValue && rawValue <= 200) {
+		result.name = 'battery'
+		result.translatable = true
+		result.value = Math.round(rawValue / 2)
+		result.descriptionText = "${device.displayName} battery was ${result.value}%"
+	}
+
+	log.debug "${device.displayName} battery was ${result.value}%"
+	result
+}
+
+private Map getBatteryResult(rawValue) {
+	log.debug 'Battery'
+	def linkText = getLinkText(device)
+
+	def result = [:]
+
+	def volts = rawValue / 10
+	if (!(rawValue == 0 || rawValue == 255)) {
+		def minVolts = 2.1
+		def maxVolts = 3.0
```

Comment:
just want to check that these are the actual min & max operating voltages for this device.

Reply:
There is no min & max operating voltages for this device.So just the 0x0021 attribute can be used for getting battery value, I will delete it .